File: java/src/leetcode/easy/MaximumSubArray.java
package leetcode.easy;

/**
 * Question: https://leetcode.com/problems/maximum-subarray/description/
 */
public class MaximumSubArray {
    public int maxSubArray(int[] nums) {
        int max = Integer.MIN_VALUE, runningMax = 0;
        for (int i = 0; i < nums.length; i++) {
            runningMax = Math.max(nums[i], runningMax + nums[i]);
            max = Math.max(max, runningMax);
        }
        return max;
    }
}

File: ruby/easy/maximum_sub_array.rb
# @param {Integer[]} nums
# @return {Integer}
def max_sub_array(nums)
    running, max_sum = 0, -((2 ** 31) - 1)
    nums.each do |num|
        running = [num, running+num].max
        max_sum = [running, max_sum].max
    end
    return max_sum
end
